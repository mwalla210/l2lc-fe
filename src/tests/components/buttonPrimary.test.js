import React from 'react'
import renderer from 'react-test-renderer'
import ButtonPrimary from '../../components/buttonPrimary'
jest.mock('../../components/buttonDefault')

const defaultOptions = {
  onClick: jest.fn(),
  disabled: false,
  text: 'text',
  className: 'className',
  type: 'primary',
  style: {}
}

describe('ButtonPrimary', () => {
  it ('Renders with snapshot (disabled: false)', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <ButtonPrimary {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders with snapshot (disabled: true)', () => {
    let options = Object.assign({}, defaultOptions)
    options.disabled = true
    const component = renderer.create(
      <ButtonPrimary {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders with snapshot (disabled: true, className: empty string)', () => {
    let options = Object.assign({}, defaultOptions)
    options.className = ''
    const component = renderer.create(
      <ButtonPrimary {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders with snapshot (disabled: false, className: empty string)', () => {
    let options = Object.assign({}, defaultOptions)
    options.className = ''
    const component = renderer.create(
      <ButtonPrimary {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
