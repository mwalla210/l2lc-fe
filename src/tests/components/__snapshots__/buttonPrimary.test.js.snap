// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimary Renders with snapshot (disabled: false) 1`] = `null`;

exports[`ButtonPrimary Renders with snapshot (disabled: false, className: empty string) 1`] = `null`;

exports[`ButtonPrimary Renders with snapshot (disabled: true) 1`] = `null`;

exports[`ButtonPrimary Renders with snapshot (disabled: true, className: empty string) 1`] = `null`;
